make/tests

```diff
@@ -1,67 +1,57 @@
 N_TESTS ?= 100 # override this in make/local. If <= 0, N_TESTS + 1 is interpreted as the number of batches to group the tests into
 
 ##
-# LIBGTEST is the google test library
 # GTEST_MAIN is the file that contains the google test
 ##
-LIBGTEST = test/libgtest.a
 GTEST_MAIN = $(GTEST)/src/gtest_main.cc
-CFLAGS_GTEST += -isystem $(GTEST)/include -isystem $(GTEST)
+GTEST_CXXFLAGS += -isystem $(GTEST)/include -isystem $(GTEST) -O$O
 
 ##
 # Build the google test library.
 ##
-$(LIBGTEST): $(LIBGTEST)(test/gtest.o)
+$(GTEST)/src/gtest-all.o: CXXFLAGS += $(GTEST_CXXFLAGS)
 
-test/gtest.o: $(GTEST)/src/gtest-all.cc
-	@mkdir -p test
-	$(COMPILE.cc) -O$O $(CFLAGS_GTEST) $< $(OUTPUT_OPTION)
+test/%$(EXE) : CXXFLAGS += $(GTEST_CXXFLAGS)
```

Comment:
That's what I was thinking - doesn't that kind of obviate the need for a specific `GTEST_CXXFLAGS` in this makefile? I guess I'm imagining that with this change, everything built in this repo will now have these flags since it will either be a test executable or one of its dependencies. (As far as upstream goes, I think we'd only include this file from the equivalent test makefile upstream so that should also be safe from contamination).

Slightly more broadly, I wonder if we actually need gtest specific `CXXFLAGS` since all of them are either preprocessor directives that just define certain gtest specific flags like `-DGTEST_HAS_PTHREAD=0` or extra include directories (which I think don't interfere or reduce performance, but I might be wrong here)... Thoughts?

PS I think this is good to go overall (once tests are passing), I'm just (maybe inappropriately) using this as a conversational focal point for a broader discussion of refactoring these makefiles, which I think we're both interested in (but correct me if I'm wrong).